src/main/java/com/epam/test_generator/entities/api/CaseTrait.java

```diff
@@ -0,0 +1,24 @@
+package com.epam.test_generator.entities.api;
+
+import com.epam.test_generator.entities.Step;
+import com.epam.test_generator.services.exceptions.BadRequestException;
+import java.util.List;
+import java.util.Objects;
+
+public interface CaseTrait {
+
+    List<Step> getSteps();
+
+
+    default Step hasStep(Step step) {
+        if (getSteps() == null || !getSteps().contains(step)) {
+            throw new BadRequestException();
+        }
+        return step;
+    }
+
+    default boolean removeStep(Step step) {
+        return Objects.requireNonNull(getSteps()).remove(hasStep(step));
```

Comment:
The code fragment looks cool but it isn't readable if taken away from class (result of hasStep non-boolean return value). Following pomkine's comment let's try to avoid defensive null checks.

    return getSteps().remove(step);